wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/linking/expression/package.scala

```diff
@@ -38,14 +38,21 @@ package object expression {
     override def consumedValueHooks(a: ArrayLiteral): Set[UnlinkedConsumedValueHook] = a.elements.toSet.flatMap { e: ExpressionElement => e.consumedValueHooks }
   }
 
+  implicit val identifierMemberAccessUnlinkedValueConsumer: UnlinkedValueConsumer[IdentifierMemberAccess] = new UnlinkedValueConsumer[IdentifierMemberAccess] {
+    override def consumedValueHooks(a: IdentifierMemberAccess): Set[UnlinkedConsumedValueHook] = Set(UnlinkedCallOutputOrIdentifierAndMemberAccessHook(a.first, a.second))
+  }
+
   implicit val expressionElementUnlinkedValueConsumer: UnlinkedValueConsumer[ExpressionElement] = new UnlinkedValueConsumer[ExpressionElement] {
     override def consumedValueHooks(a: ExpressionElement): Set[UnlinkedConsumedValueHook] = a match {
       case _: PrimitiveLiteralExpressionElement | _: StringLiteral => Set.empty
-      case id: IdentifierLookup => id.consumedValueHooks
-      case o: ObjectLiteral => o.consumedValueHooks
-      case p: PairLiteral => p.consumedValueHooks
+      case a: ObjectLiteral => a.consumedValueHooks
+      case a: PairLiteral => a.consumedValueHooks
       case a: ArrayLiteral => a.consumedValueHooks
-      case m: MapLiteral => m.consumedValueHooks
+      case a: MapLiteral => a.consumedValueHooks
+
+      case a: IdentifierLookup => a.consumedValueHooks
+      case a: IdentifierMemberAccess => a.consumedValueHooks
```

Comment:
TOL I don't have enough background to judge in this particular case, but it seems that while typeclasses are a nice way to add type-specific handling for nonessential concerns, they do introduce quite a bit of boilerplate as evidenced above. If these actually are essential concerns they might be less noisily modeled in a supertrait, but as I said I can't really judge how essential this stuff is to the `*Literal`s.

Reply:
It's hard to tell with an `n` of 1 whether this will really work... but I'm hoping that this scheme will make switching out evaluators for different language versions easier (because they're not baked into the case classes but implemented by the specific language version).